tests/virt_autotest/login_console_as_kvm_or_xen_role.pm

```diff
@@ -0,0 +1,66 @@
+#
```

Comment:
Does the file name match the fact? I remember you share that tumbleweed does not have kvm or xen role, and this file is the login_console for tumbleweed? BTW making login_console for sle and tumbleweed in two complete files will miss the advantage of reusing the code pieces that work for both, eg some ipmi consolidation code. Is it possible to make them share some common part, and making the separate files with differentiated parts only?

Reply:
based on my reply https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12912/files#r673048301, separate the script to two parts? ie. put only one line(use_ssh_serial_console) in login_console_xx.pm, and put zypper install kvm/xen patterns to another pm?

Reply:
Let me explain more and give some feasible suggestions IMHO, based on comments about login_console_as_xx from pavel, oliver  and me.
- This login_console_xx file is to handle the process from direct power on/reboot, until the system finally boots into the wanted os via grub, then make the root-ssh console as the current active console. Suggeted action: so for both tumbleweed and sle, it is better to treat this process in this SHARED file. 1) There can be if else to differntiate, eg sle has xen/kvm role selection during grub2 while tumbleweed can directly go to the default menuentry. 2)And there are shared parts, eg select sol console at the beginning to watch grub2, as well as use_ssh_serial_console in the end, and other minor pieces.
- But all the other logics should be put in other files with names OTHER than login_console_xx(it does sth like VT host env setup, similar with what our install_package do(install softwares for vt), but more than that of course).
To be explict, the major parts in this file login_console_as_kvm_or_xen_role.pm is better to be put in another name file, eg setup_tumbleweed_vt_software_and_services(just an example): install xx_server/xx_tools packages, handling for libvirt services, and later sbin links.
- Oliver also suggests to treat all above(tw os boot, login and setup to run as kvm/xen) in a way that tw real users would do. I guess you agree with too. IMHO what you do maybe the most popular way already. But I do not have too much experience on tw, you'd better check it in tw document and community posts to double confirm.

Reply:
> * This login_console_xx file is to handle the process from direct power on/reboot, until the system finally boots into the wanted os via grub, then make the root-ssh console as the current active console. 

No. pxe menu and grub has already been done by previous step: reconnect_mgmt_console.pm & first_boot.pm in openSUSE installation. Those are normal steps in O3.

login_console_as_kvm_or_xen_role.pm here just switch to ssh console. It has no login function. I should rename it to install_kvm_or_xen role(or other names).

Reply:
Exactly. Besides tw does not have kvm or xen ROLE. Please do NOT include role keyword in the new name either.

Reply:
changed to switch_to_ssh_and_install_hypervisor.pm.